lib/spack/spack/test/cmd/stage.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import pytest
+from spack.main import SpackCommand
+import spack.environment as ev
+import spack.repo
+from spack.version import Version
+
+
+stage = SpackCommand('stage')
+env = SpackCommand('env')
+
+pytestmark = pytest.mark.usefixtures('install_mockery', 'mock_packages')
+
+
+def test_stage_spec(monkeypatch):
+    """Verify that staging specs works."""
+
+    expected = set(['trivial-install-test-package', 'mpileaks'])
+
+    def fake_stage(pkg, mirror_only=False):
+        expected.remove(pkg.name)
+
+    monkeypatch.setattr(spack.package.PackageBase, 'do_stage', fake_stage)
+
+    stage('trivial-install-test-package', 'mpileaks')
+
+    assert len(expected) == 0
```

Comment:
That's a fun way of mocking this test to only test the command, not the underlying functionality. +1